### PR TITLE
[WIP] Add Series.__array__ protocol

### PR DESCRIPTION
**Pull request by mrocklin**
@mrocklin please type a PR description here.

### CODE REVIEW

**Comment by kkraus14** [comment]
rerun tests

**Comment by harrism** [comment]
@mrocklin needs changelog

**Comment by kkraus14** [comment]
rerun tests

**Comment by kkraus14** [comment]
Looks like there are tests that this change causes to fail. (Likely logic in the test)

**Comment by kkraus14** [comment]
@mrocklin I'm going to take this over for 0.9 as most of the changes needed to support this will likely be in fixing tests that constructed numpy arrays from cudf columns in bad ways.

File: python/cudf/dataframe/series.py [hunk 1]
```diff
@@ -240,6 +240,12 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
         else:
             return NotImplemented
 
+    def __array__(self, dtype=None, **kwargs):
+        out = self.to_array()
+        if dtype and out.dtype != dtype:
+            out = out.astype(dtype)
```
Comment (kkraus14) [inline python/cudf/dataframe/series.py]: Note to self, typecast on the GPU before converting to CPU array as it will be faster.

**Comment by kkraus14** [comment]
Going to close this as it's stale. Will follow up with a separate PR.

**Pull request closed without merging**